test/runtime.spec.ts

```diff
@@ -212,3 +212,58 @@ describe('map', () => {
     return true;
   });
 });
+
+describe('getAll', () => {
+  jsc.property('returns empty array when no matches', jsc.json, async dict => {
+    const items = await getAll(dict, (_n: any): _n is string => false);
+    expect(items).toEqual([]);
+    return true;
+  });
+
+  jsc.property('returns full object in array when already matches root', jsc.json, async dict => {
+    const items = await getAll(dict, (_n: any): _n is string => true);
+    expect(items).toEqual([dict]);
+    return true;
+  });
+
+  jsc.property(
+    'matches array filter with one dimensional arrays',
+    jsc.array(jsc.oneof([jsc.asciistring, jsc.integer, jsc.bool])),
+    async arr => {
+      const items = await getAll(arr, (n: any): n is string => _.isString(n));
+      expect(items).toEqual(arr.filter(item => _.isString(item)));
+      return true;
+    }
+  );
+
+  jsc.property(
+    'matches Object.values.filter with one dimensional objects',
+    jsc.dict(jsc.oneof([jsc.asciistring, jsc.integer, jsc.bool])),
+    async dict => {
+      const items = await getAll(dict, (n: any): n is string => _.isString(n));
+      expect(items).toEqual(Object.values(dict).filter(item => _.isString(item)));
+      return true;
+    }
+  );
+
+  jsc.property(
+    'matches single primitives',
+    jsc.oneof([jsc.asciistring, jsc.integer, jsc.bool]),
+    async item => {
+      const items = await getAll(item, (n: any): n is string => _.isString(n));
+      expect(items).toEqual(_.isString(item) ? [item] : []);
+      return true;
+    }
+  );
+
+  jsc.property('returns all nested values with the property', jsc.json, async value => {
+    const predicate = (n: any): n is string => typeof n === 'string';
```

Comment:
this does not catch the parent and child both matching as this does not actually test nested structure (a string cant have a child). If you change the predicate to ´typeof n === 'object´ I think it might flag it.

Reply:
Excellent catch, thank you! :clap: